operator/src/test/java/oracle/kubernetes/operator/steps/ServerDownIteratorStepTest.java

```diff
@@ -0,0 +1,383 @@
+
+    testSupport.defineResources(domain);
+    testSupport
+            .addToPacket(ProcessingConstants.DOMAIN_TOPOLOGY, domainConfig)
+            .addDomainPresenceInfo(domainPresenceInfo);
+    testSupport.addComponent(
+            ProcessingConstants.PODWATCHER_COMPONENT_NAME,
+            PodAwaiterStepFactory.class,
+            new PodHelperTestBase.DelayedPodAwaiterStepFactory(1));
+
+  }
+
+  /**
+   * Cleanup env after tests.
+   * @throws Exception if test support failed
+   */
+  @After
+  public void tearDown() throws Exception {
+    for (Memento memento : mementos) {
+      memento.revert();
+    }
+
+    testSupport.throwOnCompletionFailure();
+  }
+
+  @Test
+  public void withConcurrencyOf1_bothClusteredServersShutdownSequentially() {
+    configureCluster(CLUSTER).withMaxConcurrentShutdown(1).withReplicas(1);
+    addWlsCluster(CLUSTER, 8001, MS1, MS2);
+    domainPresenceInfo = createDomainPresenceInfoWithServers(MS1, MS2);
+    testSupport
+            .addDomainPresenceInfo(domainPresenceInfo);
+
+    invokeStepWithServerShutdownInfos(createServerShutdownInfosForCluster(CLUSTER,MS1, MS2));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.FALSE));
+    testSupport.setTime(10, TimeUnit.SECONDS);
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.TRUE));
+  }
+
+  @Test
+  public void withConcurrencyOf2_bothClusteredServersShutdownConcurrently() {
+    configureCluster(CLUSTER).withMaxConcurrentShutdown(2).withReplicas(1);
+    addWlsCluster(CLUSTER, PORT, MS1, MS2);
+    domainPresenceInfo = createDomainPresenceInfoWithServers(MS1, MS2);
+    testSupport
+            .addDomainPresenceInfo(domainPresenceInfo);
+
+    invokeStepWithServerShutdownInfos(createServerShutdownInfosForCluster(CLUSTER,MS1, MS2));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.TRUE));
+  }
+
+  @Test
+  public void withConcurrencyOf0_clusteredServersShutdownConcurrently() {
+    configureCluster(CLUSTER).withMaxConcurrentShutdown(0);
+    addWlsCluster(CLUSTER, PORT, MS1, MS2);
+
+    domainPresenceInfo = createDomainPresenceInfoWithServers(MS1, MS2);
+    testSupport
+            .addDomainPresenceInfo(domainPresenceInfo);
+
+    invokeStepWithServerShutdownInfos(createServerShutdownInfosForCluster(CLUSTER,MS1, MS2));
+
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS2), is(Boolean.TRUE));
+    MatcherAssert.assertThat(
+            domainPresenceInfo.isServerPodBeingDeleted(MS1), is(Boolean.TRUE));
+  }
+
+  @Test
+  public void withReplicaCountOf0AndConcurrencyOf1_clusteredServersShutdownConcurrently() {
```

Comment:
What is going on in this test? Why do the servers shut down concurrently with a concurrency of 1? There seems to be something important going on, and the name of the test doesn't clarify it.

Reply:
How about `whenClusterShutdown_concurrencySettingIsIgnored` ? (replica count set to 0 means the cluster is shutting down, which is why we don't care about concurrency)